NetCoreMentoring/src/NetCoreMentoring.App/Infrastructure/ImageCacheMiddleware.cs

```diff
@@ -0,0 +1,80 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Configuration;
+using NetCoreMentoring.Core;
+using NetCoreMentoring.Core.Utilities;
+
+namespace NetCoreMentoring.App.Infrastructure
+{
+    // I use caching with help of attribute instead
+    // Don't delete this just for completing homework module
+    public class ImageCacheMiddleware
+    {
+        private readonly RequestDelegate next;
+        private static readonly Dictionary<string, string> SupportedPictureContentTypes = new Dictionary<string, string>()
+        {
+            {"image/jpeg", "jpeg"},
+            {"image/png", "png"}
+        };
+        private readonly string CacheImagePath;
+        private readonly IConfiguration _configuration;
+
+        public ImageCacheMiddleware(RequestDelegate next, IConfiguration configuration)
+        {
+            this.next = next;
+            _configuration = configuration;
+
+            CacheImagePath = configuration["CacheImagePath"];
+        }
+
+        public async Task Invoke(HttpContext context)
+        {
```

Comment:
consider this logic instead:

                
                if (_cachingOptions.CachedCodePath.TryGetValue(context.Request.Path, out var cachedKey) &&
                    context.Request.Query.TryGetValue(cachedKey, out var keyValue))
                {
                    if (IsPictureCached(keyValue, out var path))
                    {
                        // return cached pictures
                        var bytes = File.ReadAllBytes(path);
                        await context.Response.Body.WriteAsync(bytes, 0, bytes.Length);
                        context.Response.ContentType = "image/png";
                        return;
                    }
                    else
                    {
                        // todo logic to save images into cache
                    }
                }

                // call next steps
                await _next.Invoke(context);

Reply:
done